### PR TITLE
[remote-executor] Isolate wormhole-solana crate

### PR DESCRIPTION
**Pull request by guibescos**
The goal of this PR is :
- Make remote-executor use the new version of `wormhole-sdk`
- Move the `wormhole-solana` (that wormhole doesn't have) to its own repo and update it to use `wormhole-sdk`

Wormhole team needs to :
- Fix the crate patch issue with serde_wormhole
- Decide what happens to `wormhole-solana`

### CODE REVIEW

File: governance/remote_executor/rust-toolchain.toml
Comment (guibescos) [inline governance/remote_executor/rust-toolchain.toml:2]: Drive-by pin blockchain

File: governance/remote_executor/programs/remote-executor/src/tests/executor_simulator.rs
Comment (guibescos) [inline governance/remote_executor/programs/remote-executor/src/tests/executor_simulator.rs:90]: Newer versions of solana can't find the binary with the previous logic

File: .github/workflows/remote-executor.yml
Comment (guibescos) [inline .github/workflows/remote-executor.yml:23]: I had to bump this

File: governance/remote_executor/Cargo.toml
Comment (guibescos) [inline governance/remote_executor/Cargo.toml:11]: This patch.crate-io wormhole thing is getting out of control, wormhole needs to fix

File: wormhole_attester/Cargo.lock
Comment (guibescos) [inline wormhole_attester/Cargo.lock:1]: Attester cargo lock updated since attester governance cli uses remote-executor

File: governance/remote_executor/programs/remote-executor/src/tests/executor_simulator.rs
Comment (Reisen) [reply governance/remote_executor/programs/remote-executor/src/tests/executor_simulator.rs:90]: Can you explain why?

File: governance/remote_executor/programs/remote-executor/Cargo.toml
Comment (Reisen) [inline governance/remote_executor/programs/remote-executor/Cargo.toml:21]: We should really get away from this, maybe we can chat about what needs to be upstreamed. We're running into this in pythnet too .

File: governance/remote_executor/cli/src/main.rs
Comment (Reisen) [inline governance/remote_executor/cli/src/main.rs:98]: Nice nice nice.

File: governance/remote_executor/Cargo.toml
Comment (Reisen) [reply governance/remote_executor/Cargo.toml:11]: Agreed, reasonable fix for now though. Let's ask for publishing.

File: governance/remote_executor/programs/remote-executor/Cargo.toml
Comment (guibescos) [reply governance/remote_executor/programs/remote-executor/Cargo.toml:21]: This is a temporary home.

File: governance/remote_executor/programs/remote-executor/src/tests/executor_simulator.rs
Comment (guibescos) [reply governance/remote_executor/programs/remote-executor/src/tests/executor_simulator.rs:90]: Previous versions "1.13.x" would set an environment variable BPF_OUT_DIR and that's how the code would find the binary.
New versions "1.14.x" don't seems to set it, so `find_file` fails.

Comment (Reisen) [reply governance/remote_executor/programs/remote-executor/src/tests/executor_simulator.rs:90]: Yeah I recall the `BPF_OUT_DIR` issues as well. Can you mention what might be the reason it's not doing it before we merge this? Don't worry if it takes too much time you can merge over this commit, it would just be nice to know what changed here for other solana related work.

**Review by Reisen — APPROVED** [review]

**Pull request merged**
